packages/fx-core/src/core/FxCore.ts

```diff
@@ -1920,6 +1920,16 @@ export class FxCore {
       );
       const listResult = await specParser.list();
       const authApis = listResult.APIs.filter((value) => value.isValid && !!value.auth);
+      if (
+        inputs.platform === Platform.VSCode &&
+        featureFlagManager.getBooleanValue(FeatureFlags.KiotaIntegration) &&
+        inputs[QuestionNames.ApiPluginType] === ApiPluginStartOptions.apiSpec().id &&
```

Comment:
"inputs[QuestionNames.ApiPluginType] === ApiPluginStartOptions.apiSpec().id" is not needed. This block is in if isGenerateFromApiSpec